### PR TITLE
get_projects: try openind if oidc fails for the OS-FEDERATION path

### PR DESCRIPTION
**Pull request by gwarf**
try openind if oidc fails for the OS-FEDERATION path

### CODE REVIEW

File: get-projects.py [hunk 1]
```diff
@@ -37,18 +39,28 @@ def get_keystone_url(os_auth_url, path):
...
         os_auth_url,
-        "/v3/OS-FEDERATION/identity_providers/egi.eu/protocols/oidc/auth")
+        "/v3/OS-FEDERATION/identity_providers/egi.eu/protocols/%s/auth" %
+        method)
     r = requests.post(url,
                       headers={'Authorization': 'Bearer %s' % access_token})
-    return r.headers['X-Subject-Token']
+    if r.status_code != requests.codes.created:
+        if method != 'openid':
+            return get_unscoped_token(os_auth_url, access_token, 'openid')
```
Comment (enolfc) [inline get-projects.py]: This recursive call feels a bit weird to me and makes the function harder to test (if we were doing unit testing). 

Could you raise here an exception and do the re-try with another method outside?

**Review by enolfc — CHANGES_REQUESTED** [review]
Thanks @gwarf, see the comments.

If we had the egi client...

File: get-projects.py (hunk 1 above)
Comment (gwarf) [reply get-projects.py]: Done.

**Review by enolfc — APPROVED** [review]
Looks good! Thanks @gwarf

**Pull request merged**